src/exports.rs

```diff
@@ -87,20 +83,16 @@ pub fn do_query<T: DeserializeOwned + JsonSchema>(
 }
 
 fn _do_init<T: DeserializeOwned + JsonSchema>(
-    init_fn: &dyn Fn(
-        &mut Extern<ExternalStorage, ExternalApi>,
-        Params,
-        T,
-    ) -> Result<Response, Error>,
-    params_ptr: *mut c_void,
+    init_fn: &dyn Fn(&mut Extern<ExternalStorage, ExternalApi>, Env, T) -> Result<Response, Error>,
+    env_ptr: *mut c_void,
     msg_ptr: *mut c_void,
 ) -> Result<*mut c_void, Error> {
-    let params: Vec<u8> = unsafe { consume_region(params_ptr)? };
+    let env: Vec<u8> = unsafe { consume_region(env_ptr)? };
     let msg: Vec<u8> = unsafe { consume_region(msg_ptr)? };
-    let params: Params = from_slice(&params).context(ParseErr { kind: "Params" })?;
+    let env: Env = from_slice(&env).context(ParseErr { kind: "Env" })?;
```

Comment:
I hate how Rust allows variable shadowing. This is very dangerous in refactorings. Anyways, unrelated of this PR.